src/extractor/scripting_environment_lua.cpp

```diff
@@ -598,6 +615,9 @@ void Sol2ScriptingEnvironment::ProcessTurn(ExtractionTurn &turn)
         {
             turn_function(turn);
 
+            if( turn.weight == MAX_TURN_WEIGHT )
+              turn.weight = context.properties.GetMaxTurnWeight();
```

Comment:
It is better to set here `std::min(turn.weight, context.properties.GetMaxTurnWeight())` to avoid `MAX_TURN_WEIGHT` and print later number of clamped values